### DISCUSSION TITLE
Error when developing WSL plugin in Rust

### DISCUSSION THREAD

**Discussion by mveril** [microsoft/WSL#11914] [General] [answered]
Hello,

I am currently developing a [Rust framework for WSL plugin development](https://github.com/mveril/wslplugins-rs), building upon the [wsl-plugin-sample](https://github.com/microsoft/wsl-plugin-sample). Despite following the setup, I've encountered an error when installing the sample Rust plugin:

```
Error code: Wsl/Service/CreateInstance/CreateVm/Plugin/0x8007007f
```

I'm having trouble debugging this error. Normally, the entry point should be well defined and I expected some logs to be generated in the specified log file. However, it appears that the WSL service does not recognize the entry point of my plugin.

Here is the entry point definition:

```rust
#[no_mangle]
pub extern "C" fn WSLPLUGINAPI_ENTRYPOINTV1(api: *const WSLPluginAPIV1, hooks: *mut WSLPluginHooksV1) -> HRESULT {
    create_plugin(api, hooks).into()
}
```

My project is open source, and I encourage you to review the code [here](https://github.com/mveril/wslplugins-rs) if the information provided is insufficient.

Could you assist in resolving this issue? Any guidance or suggestions would be greatly appreciated.

Thank you in advance for your assistance.

**Comment by elsaco** [comment]
After building, signing, and registering the plugin `PATH _NOT_FOUND` error is thrown when launching WSL:

```
A fatal error was returned by plugin 'wsl-plugin-rs'
Error code: Wsl/Service/CreateInstance/CreateVm/Plugin/ERROR_PATH_NOT_FOUND

[process exited with code 4294967295 (0xffffffff)]
You can now close this terminal with Ctrl+D, or press Enter to restart.
```

**Reply by mveril** [reply]
Thanks for your response @elsaco. I reproduced your issue by changing the registry value of the plugin.
Can you check your registry at `HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\Lxss\Plugins` please ? As described on the readme of the [wsl-plugin-sample](https://github.com/microsoft/wsl-plugin-sample)., The value of `wsl-plugin-rs` at this key should be the full path of the dll.

**Comment by mveril** [comment, marked as answer]
I solved the issue. In the header file, the entry point is defined as:
```c
#define WSLPLUGINAPI_ENTRYPOINTV1 WSLPluginAPIV1_EntryPoint
```
Thus, the actual name of the entry point is `WSLPluginAPIV1_EntryPoint` not `WSLPLUGINAPI_ENTRYPOINTV1`.
The name is replaced by the preprocessor using the macro.
So in the context of my rust code I need to use `WSLPluginAPIV1_EntryPoint`.
